eng/common/pipelines/ci.yml

```diff
@@ -0,0 +1,107 @@
+trigger: none
+pr:
+  branches:
+    include:
+      - checkEnforcer
+
+extends:
+  template: /eng/common/pipelines/templates/1es-redirect.yml
+  parameters:
+    stages:
+      - stage: InitStage
+        displayName: Initialize
+        pool:
+          name: $(LINUXPOOL)
+          image: $(LINUXVMIMAGE)
+          os: linux
+        jobs:
+          - job: InitJob
+            displayName: Initialize
+            steps:
+              - pwsh: |
+                  Write-Host "Setting RunCore to true"
+                  Write-Host "##vso[task.setvariable variable=RunCore;isOutput=true]true"
+                  Write-Host "Setting RunCSharp to true"
+                  Write-Host "##vso[task.setvariable variable=RunCSharp;isOutput=true]true"
+                  Write-Host "Setting RunPython to true"
+                  Write-Host "##vso[task.setvariable variable=RunPython;isOutput=true]true"
+                name: InitStep
+                displayName: Initialize
+
+      # Run csharp stages if RunCSharp == true
+      - template: /packages/http-client-csharp/eng/pipeline/templates/ci-stages.yml
+        parameters:
+          DependsOn: InitStage
+          Condition: eq('true', stageDependencies.InitStage.outputs['InitJob.InitStep.RunCSharp'])
+
+      # Run core stages if RunCore == true
+      - stage: Build_Core
```

Comment:
should that just be a call to a template instead?

Reply:
yes, we are going to do that as a follow up step https://github.com/microsoft/typespec/issues/3117